pydough/unqualified/qualification.py

```diff
@@ -351,6 +355,192 @@ def qualify_collation(
         )
         return CollationExpression(qualified_expr, asc, na_last)
 
+    def parse_general_condition(self, access: SubCollection) -> None:
+        """
+        Parses a general join condition of a sub-collection access from a
+        PyDough string into an unqualified node and then into a QDAG node
+        containing sided references. Stores the answer inside the
+        sub-collection access.
+
+        Args:
+            `access`: the sub-collection access containing the join condition
+            that is being parsed.
+        """
+        assert isinstance(access.subcollection_property, GeneralJoinMetadata)
+        property: GeneralJoinMetadata = access.subcollection_property
+        self_name: str = property.self_name
+        other_name: str = property.other_name
```

Comment:
I assume that when there are reverse relationships, the join condition has `self` and `other` reversed right?

Reply:
Discard this, saw the other comment.